graph_weather/models/atmorep/data/dataset.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+from torch.utils.data import Dataset
+
+from graph_weather.models.atmorep.config import AtmoRepConfig
+
+
+class ERA5Dataset(Dataset):
```

Comment:
For this, we could probably just run from ARCO-ERA5, it might be easier. Ideally, this would also be a more separate PR for the dataset, under `/data/` and be more general than AtmoRep-specific.